homeassistant/components/honeywell/climate.py

```diff
@@ -34,266 +38,229 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_USERNAME): cv.string,
     vol.Required(CONF_PASSWORD): cv.string,
-    vol.Optional(CONF_AWAY_TEMPERATURE,
-                 default=DEFAULT_AWAY_TEMPERATURE): vol.Coerce(float),
     vol.Optional(CONF_COOL_AWAY_TEMPERATURE,
                  default=DEFAULT_COOL_AWAY_TEMPERATURE): vol.Coerce(int),
     vol.Optional(CONF_HEAT_AWAY_TEMPERATURE,
                  default=DEFAULT_HEAT_AWAY_TEMPERATURE): vol.Coerce(int),
     vol.Optional(CONF_REGION, default=DEFAULT_REGION): vol.In(REGIONS),
 })
 
+HVAC_MODE_TO_HW_MODE = {
+    'SwitchOffAllowed': {HVAC_MODE_OFF: 'off'},
+    'SwitchAutoAllowed': {HVAC_MODE_HEAT_COOL: 'auto'},
+    'SwitchCoolAllowed': {HVAC_MODE_COOL: 'cool'},
+    'SwitchHeatAllowed': {HVAC_MODE_HEAT: 'heat'},
+}
+HW_MODE_TO_HVAC_MODE = {
+    'off': HVAC_MODE_OFF,
+    'emheat': HVAC_MODE_HEAT,
+    'heat': HVAC_MODE_HEAT,
+    'cool': HVAC_MODE_COOL,
+    'auto': HVAC_MODE_HEAT_COOL,
+}
+HW_MODE_TO_HA_HVAC_ACTION = {
+    'off': CURRENT_HVAC_OFF,
+    'fan': CURRENT_HVAC_IDLE,
+    'heat': CURRENT_HVAC_HEAT,
+    'cool': CURRENT_HVAC_COOL,
+}
+FAN_MODE_TO_HW = {
+    'fanModeOnAllowed': {FAN_ON: 'on'},
+    'fanModeAutoAllowed': {FAN_AUTO: 'auto'},
+    'fanModeCirculateAllowed': {FAN_DIFFUSE: 'circulate'},
+}
+HW_FAN_MODE_TO_HA = {
+    'on': FAN_ON,
+    'auto': FAN_AUTO,
+    'circulate': FAN_DIFFUSE,
+    'follow schedule': FAN_AUTO,
+}
+
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Honeywell thermostat."""
     username = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
-    region = config.get(CONF_REGION)
 
-    if region == 'us':
-        return _setup_us(username, password, config, add_entities)
+    if config.get(CONF_REGION) == 'us':
+        try:
+            client = somecomfort.SomeComfort(username, password)
+        except somecomfort.AuthError:
+            _LOGGER.error("Failed to login to honeywell account %s", username)
+            return False
```

Comment:
Nothing is checking this return value. Just `return`.

Reply:
done